### DISCUSSION TITLE
Join the Frictionless Data Hackathon

### DISCUSSION THREAD

**Discussion by lwinfree** [frictionlessdata/frictionlessdata.io#693] [General]
# Hi everyone! We are going to have a hackathon on 7-8 October and you are invited!

> Register your project here: https://forms.gle/brYDcP2Fugn74fgN6

## What’s a hackathon?
You’ll work within a group of other Frictionless users to create new project prototypes based on existing Frictionless open source code. For example, use the new [Livemark](https://livemark.frictionlessdata.io/) tool to create websites that display data-driven storytelling, or use [Frictionless React Components](https://components.frictionlessdata.io/) to add data validation to your application.

## Who should participate in this hackathon?
We’re looking for contributions of all sizes and skill levels! Some skills that you would bring include: coding in Python (other languages supported too!), writing documentation, project management, having ideas, design skills, and general enthusiasm! You’ll be in a team, so you can learn from each other and help each other. You don’t have to be familiar with Frictionless yet - you can learn that during the event.

## Why should I participate?
First of all, it will be fun! You’ll meet other Frictionless users and learn something new. This is also an opportunity where you’ll have the uninterrupted support of the Frictionless core team to help you realize your prototype. Also, there will be prizes (details to be announced later).

## When will the hackathon occur?
The hackathon will be virtual and occur on 7-8 October. The event will start in the morning European time on 7 October, and will end in the evening European time on 8 October. This will allow people from around the world to participate during a time that works for them. We will be using Github and Zoom to coordinate and work virtually. Teams will be able to form before the event occurs so you can start coordinating early and hit the ground running.

## Sign me up!
Propose your project idea using this form: https://forms.gle/brYDcP2Fugn74fgN6. Registration for attendees will open in September. The event will be free, and we will also have some scholarships for attendees that would otherwise be unable to attend. More details coming soon!

## I have questions…
Please email us at frictionlessdata@okfn.org if you have questions or would like to support the Hackathon.

(text from https://frictionlessdata.io/hackathon/)

**Comment by augusto-herrmann** [comment]
Hi! I look forward to the Frictionless hackathon tomorrow, but one thing makes me worried. Although the description above says we would coordinate through Github and Zoom:

> We will be using Github and Zoom to coordinate and work virtually.

However, the [participation guide](https://docs.google.com/document/u/1/d/e/2PACX-1vReWY9N26SbveoCM7Ra4wEry8k7a5rCa3UzpBijfU_mmyME58DRDKmu0QUmx75mif4367IZdtLijFzO/pub) is all set around Discord.

![participation-guide](https://user-images.githubusercontent.com/1058414/136204917-dd84e755-0bab-48b1-b576-2d7284060400.png)

As I've argued before, Discord is [bad for your privacy](https://tosdr.org/en/service/536), and gets Privacy Grade E, which is the worst possible on ToS;DR:

![Discord Privacy Grade E](https://herrmann.tech/assets/images/2021/08/tosdr-discord-grade-en.gif)

I not only [disagree with Discord's Terms of Service and Privacy Policy](https://herrmann.tech/en/blog/2021/08/10/my-discord-with-discord-or-choosing-alternatives-with-better-terms-of-service.html), but also have manifested strongly, from day one, [against the move](https://github.com/frictionlessdata/forum/issues/13#issuecomment-637098565) from Gitter.im to Discord, a move which has been controversial.

We already have [a discussion here](https://github.com/frictionlessdata/project/discussions/690) about this, so the Hackathon could be the opportunity to try and create a [Matrix Space](https://matrix.org/blog/2021/05/17/the-matrix-space-beta) for Frictionless,

![matrix-space](https://user-images.githubusercontent.com/1058414/136206824-8a86bb31-9c7a-4662-9f21-9ba02b3a3849.png)

and perhaps [bridge it](https://matrix.org/bridges/) to the existing Discord "server"? In that case, I would need help, as I don't a Discord account (and do not intend to create one), nor would I have the necessary permissions on Discord to establish the bridge.

As for the video calls, we could use Zoom like we've been using in community calls, but I would rather use Jitsi Meet as it

* has all the features we need,
* there are many free to use services around (e.g. meet.jit.si),
* it's easy to split up in different rooms, as anyone can create new rooms without a login or account – just come up with a new url for the room and you're good to go, and
* it's free and open source software.

I use it every day to communicate with the team and it works very well. But if I have to use Zoom it's not a big problem.

**Comment by lwinfree** [comment]
Hi @augusto-herrmann ! We are indeed using Discord for the hackathon. It will be the way to communicate with the team. I hear your concerns, but for various reasons, we've decided that Discord makes the most sense for this event. You will need to use Discord to participate with the hackathon. I really hope that you create an account (and could delete it later if you wish) and are able to participate!

**Comment by augusto-herrmann** [comment]
Thanks, @lwinfree, but as I said, I don't agree with Discord's Terms of Service and Privacy Policy, therefore I cannot create an account.

Now is as good time as any (especially now during the hackathon many more people are able to contribute), so I've gone ahead and created a Matrix Space for Frictionless:

https://matrix.to/#/#frictionless-data:matrix.org

If anyone with access to Discord could help us bridge the community, here are the instructions:

https://t2bot.io/discord/

**What is bridging?**

[Bridges](https://matrix.org/bridges/) allow interoperability between Matrix and other services (like Discord). Anyone using any of the two services can read and write messages to each other. Think of it like mirroring the community, so that anyone can use the tool that suits them best.

**Comment by lwinfree** [comment]
Hi @augusto-herrmann! We'd really like to have the community in one place because it is easier to contact everyone if they are in one place, and so we don't have a split/disjointed community. It sounds like that shouldn't be a problem though with the bridge idea, as long as all messages sent from Matrix would also show up on Discord? Is that correct?

**Reply by augusto-herrmann** [reply]
> It sounds like that shouldn't be a problem though with the bridge idea, as long as all messages sent from Matrix would also show up on Discord? Is that correct?

Yes, that is exactly right! Bridging should make it an instant mirror of all messages. The community will definitely **not** be split up once we do this.

**Comment by lwinfree** [comment]
Cool! I'll try to look at this more next week - thanks for opening an issue :-)

**Reply by lwinfree** [reply]
FYI I'll try to get to it tomorrow, but might not have time til next week.
